pkg/dss/scd/server.go

```diff
@@ -169,41 +170,23 @@ func (a *Server) PutSubscription(ctx context.Context, req *scdpb.PutSubscription
 
 	// If this is an update, get the old version
 	var (
-		params               = req.GetParams()
-		notificationIndex    = 0
-		implicitSubscription = false
-		oldVersion           = 0
+		params = req.GetParams()
 	)
-	if params.OldVersion > 0 {
-		//TODO: This needs to happen in a single transaction
-		oldSub, err := a.Store.GetSubscription(ctx, id, owner)
-		if err != nil {
-			return nil, err //TODO: Change to 409 if Subscription didn't already exist
-		}
-		if oldSub.Version != int(params.OldVersion) {
-			return nil, dsserr.VersionMismatch("old_version does not match current version")
-		}
-		notificationIndex = oldSub.NotificationIndex
-		implicitSubscription = oldSub.ImplicitSubscription
-		oldVersion = oldSub.Version
-	}
 
 	// Construct Subscription model
-	sub := &models.Subscription{
-		ID:                id,
-		Version:           oldVersion + 1,
-		NotificationIndex: notificationIndex,
-		Owner:             owner,
+	sub := &scdmodels.Subscription{
+		ID:      id,
+		Owner:   owner,
+		Version: scdmodels.Version(params.OldVersion),
 
 		BaseURL:              params.UssBaseUrl,
 		NotifyForOperations:  params.NotifyForOperations,
 		NotifyForConstraints: params.NotifyForConstraints,
-		ImplicitSubscription: implicitSubscription,
 	}
 	//TODO: Set StartTime, EndTime, AltitudeHi, AltitudeLo, DependentOperations
 
 	// Store Subscription model
-	sub, err := a.Store.InsertSubscription(ctx, sub, owner) //TODO: This should be UpsertSubscription
+	sub, err := a.Store.InsertSubscription(ctx, sub) //TODO: This should be UpsertSubscription
```

Comment:
Per comment, this should really be UpsertSubscription because PUT either creates or updates -- it looks like some of the logic for updating was removed above.

Reply:
Will defer this to another PR.